engine/src/main/java/org/terasology/persistence/typeHandling/coreTypes/RuntimeDelegatingTypeHandler.java

```diff
@@ -56,7 +56,12 @@ public RuntimeDelegatingTypeHandler(TypeHandler<T> delegateHandler, TypeInfo<T>
     public PersistedData serializeNonNull(T value, PersistedDataSerializer serializer) {
         // If primitive, don't go looking for the runtime type, serialize as is
         if (typeInfo.getRawType().isPrimitive()) {
-            return delegateHandler.serialize(value, serializer);
+//            LOGGER.info("DelegateHandler: " + delegateHandler);
+            if (delegateHandler != null) {
+                return delegateHandler.serialize(value, serializer);
+            }
```

Comment:
This change should not be needed since primitive types will always have a TypeHandler (thus ensuring `delegateHandler` will never be null). The problem is propagating from somewhere else. We _could_ make this change here, but this is just deferring the change elsewhere.

Reply:
Entirely agree with you. We need to figure out the source of the problem. I've logged out all the events to find the problem, then I noticed that after the `KEY_SCAPE` event some weird events just popped out. These events are represented by a character `'\u001B' 27` (which I think is a escape character), so it belongs to primitive types. However, when I logged out its respective `delegateHandler` it had a `null` value. In other words, no `TypeHandler` for this weird char. Therefore, adding the if statement is the workaround that I found.

At this picture we can see some of the event logs, including the weird one:
![image](https://user-images.githubusercontent.com/26657147/47628496-ac384d80-db14-11e8-881e-4fd561906cb8.png)

Maybe another PR to fix that problem would be great, because I still have no idea where it came from, but yeah it has nothing to do with serialization.

Reply:
I have merge this branch with `develop`, there is a chance that this problem is fixed there. We are 95 commits behind `develop`, that's bad. :/

Reply:
I need to merge this branch with `develop`, there is a chance that this problem is fixed there. We are 95 commits behind `develop`, that's bad. :/

Reply:
Yep, figured out! I'm going to implement the `CharTypeHandler` and remove this workaround. Thanks for your help! :-)